doc/specs/#980 - SnapOnOutput.md

```diff
@@ -26,24 +26,18 @@ Typical Unix terminals work differently. Rather than disabling the output, they
 `snapOnOutput` will be a profile-level `ICoreSettings` setting of type enum. It can be set to the following values:
 - `never`: new output does not cause the viewport to update to the bottom of the scroll region
 - `noSelection`: (**default**) new output causes the viewport to update to the bottom of the scroll region **IF** no selection is active
+- `atBottom`: new output causes the viewport to update **IF** the viewport is already at the virtual bottom

```

Comment:
With this setting, if I'm scrolled up in the buffer, and text is output, do I stay at the current row, or the same relative scroll offset? In other words, if I'm scrolled up with this value, and the process emits more lines, does the viewport "shift down"?

What should happen when the buffer circles? Move the visual viewport up to maintain the same content, until the visual viewport is on row 0?

Reply:
Yes please, maintain the same content on the screen. The content should not move. The scrollbar can shrink and do what it needs to do, but don't move any text around.

If the buffer gets overrun... I don't know. I would prefer that the buffer *didn't* get overrun, but if it must, I guess start deleting rows from the top. Whether you start showing blanks at the top or start pushing the content up so that you don't have some weird scrolled-past-the-top state is up to you. Hopefully this is an edge case and it can be fixed with really large or infinite buffers.

Reply:
Added. 😊 I think at this point, the user should consider using infinite scrollback.

Reply:
before that, we should consider _supporting_ infinite scrollback 😉